EIPS/eip-7896.md

```diff
@@ -0,0 +1,142 @@
+};
+
+type InterfacesCapability = {
+    [contract: `0x${string}`]: {
+        version: string;
+        spec: unknown; // version-dependent
+    };
+    optional?: boolean;
+};
+```
+
+An interface spec MUST describe how a contract will decode calldata into a series of function arguments. The interpretation of a spec as a decoding procedure depends on a version string.
+
+The two initially defined versions are `abi-v1` and `abi-v2`, corresponding to the Contract Application Binary Interface as defined by Solidity and adopted by other languages. In these cases, the spec MUST be provided as an array as described in Solidity's documentation, page "Contract ABI Specification", section "JSON". The array MAY contain event and error descriptions in addition to function descriptions, though applications SHOULD exclude irrelevant items. The v1-v2 distinction corresponds to Solidity's `pragma abicoder v1` and `pragma abicoder v2` (or `pragma experimental ABIEncoderV2`). Vyper implements `abi-v2`.
+
+Applications SHOULD mark the `interfaces` field as `optional`, so that the user is able to proceed if their wallet does not support the capability or a provided spec version. If the field is marked `optional`, the wallet MAY ignore interfaces with unsupported spec versions.
+
+Wallets SHOULD use the provided interface specs to decode the `data` of each call in the `calls` array in order to present the user with the decoded function arguments. Wallets MAY validate the provided specs against a database of verified or trusted specs.
+
+#### Example Request
+
+```JSON
+[
+  {
+    "version": "1.0",
+    "from": "0xa22cc169386b820ab57c006a5b4980add068a7eb",
+    "chainId": "0x01",
+    "calls": [
+      {
+        "to": "0xdac17f958d2ee523a2206206994597c13d831ec7",
+        "value": "0x00",
+        "data": "0xa9059cbb000000000000000000000000f0c87f351435211efa00938a33771bf38302d1f10000000000000000000000000000000000000000000000056bc75e2d63100000"
+      }
+    ],
+    "capabilities": {
+      "interfaces": {
+        "optional": true,
+        "0xdac17f958d2ee523a2206206994597c13d831ec7": {
+          "version": "abi-v1",
+          "spec": [
+            {
+              "type": "function"
+              "name": "transfer",
+              "stateMutability": "nonpayable",
+              "inputs": [
+                {
+                  "name": "to",
+                  "type": "address"
+                },
+                {
+                  "name": "value",
+                  "type": "uint256"
+                }
+              ],
+              "outputs": [],
+            }
+          ]
+        }
+      }
+    }
+  }
+]
+```
+
+### `wallet_getCapabilities`
+
+Under the `interfaces` capability, wallets MUST include a `versions` array field listing the supported interface spec versions.
+
+#### Example Response
+
+```JSON
+{
+  "0x0": {
+    "interfaces": {
+      "supported": true,
+      "versions": ["abi-v1", "abi-v2"]
+    }
+  }
+}
+```
+
+## Rationale
+
+TBD
```

Comment:
add <-- TODO --> tag for bot to flag to you to fill this up when moving past draft

Reply:
```suggestion
<-- TODO -->
```